src/botdetection/request_filters/__init__.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from enum import Enum
+
+from flask import Request
+
+import werkzeug
+
+from .._request_info import RequestInfo
+from .._request_context import RequestContext
+from . import http_accept, http_accept_encoding, http_accept_language, http_connection, http_user_agent, ip_limit
+
+
+class RequestFilter:  # pylint: disable=too-few-public-methods
```

Comment:
Just a hint / the naming is a bit misleading: A `RouteFilter` is a (is of type)  `RequestFilter` while a `PredefinedRequestFilter` is a list (a enumeration type) of functions (with identical prototype and return type).


- `PredefinedRequestFilter` --> `FilterMethods` (or similar)

To avoid a conflict with https://github.com/searxng/searxng/pull/3462 we should not change it here in this branch / we can do it later / just wanted to leaf a note while I'm in review of this PR.